### PR TITLE
[k6] fix Response.message and Params.timeout type

### PR DESCRIPTION
**Pull request by gonsuke**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
  - `Params` ... https://k6.io/docs/javascript-api/k6-net-grpc/params/
  - `Response` ... https://k6.io/docs/javascript-api/k6-net-grpc/response
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/k6/net/grpc.d.ts [hunk 1]
```diff
@@ -29,7 +29,7 @@ export interface Params {
 
     tags?: object;
 
-    timeout?: string;
+    timeout?: number;
```
Comment (na--) [inline types/k6/net/grpc.d.ts:32]: This can be a number (milliseconds) _or_ a string.

Comment (gonsuke) [reply types/k6/net/grpc.d.ts]: Ok thanks. Fixed https://github.com/DefinitelyTyped/DefinitelyTyped/pull/57368/commits/130b6f3073d5d6fbb1e69c26c86f3e32fac00eed

Comment (ppcano) [reply types/k6/net/grpc.d.ts]: @na--, also `ConnectParams.timeout` ?

File: types/k6/net/grpc.d.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ export interface Params {
 
     tags?: object;
 
-    timeout?: string;
+    timeout?: string | number;
```
Comment (ppcano) [inline types/k6/net/grpc.d.ts:32]: Please, add a test for `string`.

File: types/k6/net/grpc.d.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@
 export interface Response {
     status: number;
 
-    message: string;
+    message: object;
```
Comment (ppcano) [inline types/k6/net/grpc.d.ts:10]: Please, add a test for this change.

**Review by ppcano — CHANGES_REQUESTED** [review]
@gonsuke Thanks so much for pointing this out and the contribution. 

Here, a related PR to update the k6 docs
https://github.com/grafana/k6-docs/pull/507

File: types/k6/net/grpc.d.ts (hunk 2 above)
Comment (gonsuke) [reply types/k6/net/grpc.d.ts:32]: Sure, added a case for `string` type as well
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/57368/commits/51d9ac5e4aefd73291f2259bf15bb0ecd959347b

File: types/k6/net/grpc.d.ts (hunk 3 above)
Comment (gonsuke) [reply types/k6/net/grpc.d.ts:10]: Sorry, no idea how do I test for this. Does [this line](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/57368/files#diff-c5fc3e839decd92e84fec26d34351fdae689002a05f09e0474a999e3a8b43f6dR25) cover the case?

Comment (ppcano) [reply types/k6/net/grpc.d.ts:10]: Should work replacing the line with

```
response.message; // $ExpectType object
```

**Review by ppcano — CHANGES_REQUESTED** [review]

**Review by ppcano — CHANGES_REQUESTED** [review]
@gonsuke, 

As part of this PR, please, could you also fix [ConnectParams.timeout](https://k6.io/docs/javascript-api/k6-net-grpc/client/client-connect-address-params#connectparams)

File: types/k6/net/grpc.d.ts (hunk 3 above)
Comment (gonsuke) [reply types/k6/net/grpc.d.ts:10]: Thank you for your guidance.
 Fixed
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/57368/commits/69023ae288075b64abae852410b437965dd85273

**Comment by gonsuke** [comment]
> As part of this PR, please, could you also fix [ConnectParams.timeout](https://k6.io/docs/javascript-api/k6-net-grpc/client/client-connect-address-params#connectparams)

Sure, fixed https://github.com/DefinitelyTyped/DefinitelyTyped/pull/57368/commits/c75e5168d0adfed14e6fad9f9047c3ec82bf33a8 and https://github.com/DefinitelyTyped/DefinitelyTyped/pull/57368/commits/0497d925a5e744c60e1ae074d429030299fe7df5

**Review by ppcano — APPROVED** [review]

**Comment by ppcano** [comment]
Ready to merge

**Comment by ppcano** [comment]
@gonsuke, thanks so much for your contribution.

**Comment by gonsuke** [comment]
> @gonsuke, thanks so much for your contribution.

@ppcano 
no problem, thank you so much for your review!

**Pull request merged**